src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -193,6 +196,15 @@ public SecurityRoles getSecurityRoles(Set<String> roles) {
         return configModel.getSecurityRoles().filter(roles);
     }
 
+    public SecurityRoles getSecurityRoleForPlugin(String pluginIdentifier) {
+        SecurityRoles pluginRole = pluginRoles.get(pluginIdentifier);
+        if (pluginRole == null) {
+            pluginRole = configModel.getSecurityRoles().createSecurityRole(pluginIdentifier, Set.of(BulkAction.NAME), Map.of());
```

Comment:
I had to add Bulk in here so that writing a document to a system index passed. I will remove Bulk from here and post the error message in this comment thread. Bulk is not really an action in and of itself, Bulk is comprised of individual actions which will each get authorized individually. 

FYI I heard back from other plugins about needing to perform actions outside of the authenticated user context and have now identified 2 cases where plugins want to perform a cluster action of index operation outside of the authenticated user context.

I created a PR in core to introduce a mechanism for IdentityAwarePlugins to request to be able to perform actions and prompt cluster admins on install: https://github.com/opensearch-project/OpenSearch/pull/15778

With the changes in that PR, I plan to update this method to pass in the cluster actions and index actions that were requested by the plugin and agreed to by a cluster admin on install.